provision/pool/pool.go

```diff
@@ -189,7 +186,7 @@ func (p *Pool) allowedValues() (map[string][]string, error) {
 				validNames = append(validNames, n)
 			}
 		}
-		resolved[k] = validNames
+		resolved[string(k)] = validNames
```

Comment:
Shouldn't `allowedValues` return values change to `map[poolConstraintType][]string, error`?